src/lsp/swift-mesonlsp.ts

```diff
@@ -0,0 +1,64 @@
+import * as os from "os";
+
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  Executable,
+  TransportKind
+} from "vscode-languageclient/node";
+import {
+  ExtensionContext,
+} from "vscode"
+import { LanguageServerClient } from "../lsp";
+
+export class SwiftMesonLspLanguageClient extends LanguageServerClient {
+  ls: LanguageClient | null = null
+
+  constructor(ctx: ExtensionContext) {
+    super("Swift-MesonLSP", ctx);
+  }
+
+  get downloadInfo(): [url: string, hash: string] | null {
+    const platform = os.platform();
+    if (platform != "win32" && platform != "darwin")
+      return null;
+    const arch = os.arch();
+    if (arch != "x64")
+      return null;
+    if (platform == "win32")
+      return ["https://github.com/JCWasmx86/Swift-MesonLSP/releases/download/v2.0/Swift-MesonLSP-win64.zip",
+              "fc0d930eb67525309291907c07df8655920cdc651f17f2320d09cc5271a23876"]
+    return ["https://github.com/JCWasmx86/Swift-MesonLSP/releases/download/v2.0/Swift-MesonLSP-macos12.zip",
+            "c10999081cef8b56ea164d9a170470c268045254829e48bf4ac137a07a8a874c"]
+  }
+
+  supportsSystem(os: string, arch: string): boolean {
```

Comment:
Doesn't your language server support linux?

Reply:
Yes but no  setup here in the extension due to the issues discussed elsewhere

Reply:
I have two ideas regarding this:
- Automatically setup the package manager on supported OS. This requires elevation to root (
- Upload more binaries for different distributions to Github Releases (Compiles on Ubuntu 18.04/20.04/22.04, Fedora, Arch). But it would be outside of the package manager and at least I would be annoyed it some extension downloads some binary albeit Linux has a package manager.